ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable no-console */
+
+import dotenv from "dotenv";
+dotenv.config();
+
+import { AptosAccount, FaucetClient, AptosToken, CoinClient, Network, Provider, HexString } from "aptos";
+
+(async () => {
+  // Create API and faucet clients.
+  // :!:>section_1a
+  const network = Network.DEVNET;
+  const provider = new Provider(network);
+  const faucetClient = new FaucetClient(`https://fullnode.${network}.aptoslabs.com/v1`, `https://faucet.${network}.aptoslabs.com`); // <:!:section_1a
+
+  // Create client for working with the token module.
+  // :!:>section_1b
+  const aptosTokenClient = new AptosToken(provider); // <:!:section_1b
+
+  // Create a coin client for checking account balances.
+  const coinClient = new CoinClient(provider.aptosClient);
+
+  // Create accounts.
+  // :!:>section_2
+  const alice = new AptosAccount();
+  const bob = new AptosAccount(); // <:!:section_2
+
+  // Print out account addresses.
+  console.log("=== Addresses ===");
+  console.log(`Alice: ${alice.address()}`);
+  console.log(`Bob: ${bob.address()}`);
+  console.log("");
+
+  // Fund accounts.
+  // :!:>section_3
+  await faucetClient.fundAccount(alice.address(), 100_000_000);
+  await faucetClient.fundAccount(bob.address(), 100_000_000); // <:!:section_3
+
+  console.log("=== Initial Coin Balances ===");
+  console.log(`Alice: ${await coinClient.checkBalance(alice)}`);
+  console.log(`Bob: ${await coinClient.checkBalance(bob)}`);
+  console.log("");
+
+  console.log("=== Creating Collection and Token ===");
+
+  const collectionName = "Alice's";
+  const tokenName = "Alice's first token";
+  const maxSupply = 1;
+
+  // Create the collection.
+  // :!:>section_4
+  const txnHash1 = await aptosTokenClient.createCollection(
+    alice,
+    "Alice's simple collection",
+    collectionName,
+    "https://alice.com",
+    maxSupply,
+    {
+      royaltyNumerator: 5,
+      royaltyDenominator: 100,
+    },
+  ); // <:!:section_4
+  await provider.aptosClient.waitForTransaction(txnHash1, { checkSuccess: true });
+
+  // Create a token in that collection.
+  // :!:>section_5
+  const txnHash2 = await aptosTokenClient.mint(
+    alice,
+    collectionName,
+    "Alice's simple token",
+    tokenName,
+    "https://aptos.dev/img/nyan.jpeg",
+    [],
+    [],
+    [],
+  ); // <:!:section_5
+  await provider.aptosClient.waitForTransaction(txnHash2, { checkSuccess: true });
+
+  // Print the collection data.
+  // :!:>section_6
+  const collectionData = (await provider.getCollectionData(alice.address(), collectionName)).current_collections_v2[0];
```

Comment:
getCollectionData() is an indexer query, later in the code we check whether indexer and devnet in sync - if not we skip tests. so if they are not synced, this test would "fail". maybe best would be to check indexer and devnet chain id and then tun this example if they are synced

Reply:
that's still a thing, it is rare and with the local env should be completely gone

Reply:
ok gotcha, I fixed it